### PR TITLE
add how to use eigen cn doc

### LINKED ISSUE #3751: need a doc to show how to make tensor computation with Eigen Tensor
Since we have a doc about how to develop a new operator, a doc about how to use Eigen library to make Tensor computation is necessary.

### PR DESCRIPTION
**Pull request by QiJune**
Fix #3751

### CODE REVIEW

File: doc/howto/dev/use_eigen_cn.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+Eigen Tensor模块对element-wise计算提供了强大的支持，并且书写一份代码，可以同时在CPU、GPU执行。但Eigen Tensor是一个正在开发中的模块，因此可能测试不够完备，文档较少。
+
+关于Eigen Tensor模块的详细介绍请参考[文档](https://github.com/RLovelett/eigen/blob/master/unsupported/Eigen/CXX11/src/Tensor/README.md)
+
+
+### paddle::framework::Tensor
+
+Paddle Tensor定义在framework目录下，其主要接口如下：
+
+```
```
Comment (Canpio) [inline doc/howto/dev/use_eigen_cn.md]: 用```cpp，给代码加上语法高亮

File: doc/howto/dev/use_eigen_cn.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+for (int i = 0; i < 1 * 2 * 3; i++) {
+  p[i] = static_cast<float>(i);
+}
+
+EigenTensor<float, 3>::Type et = EigenTensor<float, 3>::From(t);
+```
+
+From是EigenTensor模板提供的一个接口，可以实现从paddle::framework::Tensor到对EigenTensor的转换。由于Tensor的rank是模板参数，因此在转换时需要显示的指定。
+
+需要额外注意的是，EigenVector<T>::From方法是把paddle中的一维Tensor转为Eigen的一维Tensor，在这里用EigenVector来表示；而EigenVector<T>::Flatten方法是把paddle中的一个Tensor进行reshape操作，压扁成为Eigen的一维Tensor，类型仍然为EigenVector。
```
Comment (Canpio) [inline doc/howto/dev/use_eigen_cn.md]: 这里可以先介绍下EigenVector，再讲它的`Flatten`和`From`的区别。还可以加个例子。

File: doc/howto/dev/use_eigen_cn.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
+## 在Paddle中如何使用Eigen
+
+神经网络本质上是一个计算图，计算需要的数据存放在`Tensor`中，而计算过程是由`Operartor`来描述的。在执行时，`Operator`调用对应`OpKernel`中的`Compute`接口，实现对`Tensor`的操作。
+
+
+### Eigen Tensor模块
+
+Eigen Tensor模块对element-wise计算提供了强大的支持，并且书写一份代码，可以同时在CPU、GPU执行。但Eigen Tensor是一个正在开发中的模块，因此可能测试不够完备，文档较少。
+
+关于Eigen Tensor模块的详细介绍请参考[文档](https://github.com/RLovelett/eigen/blob/master/unsupported/Eigen/CXX11/src/Tensor/README.md)
```
Comment (typhoonzero) [inline doc/howto/dev/use_eigen_cn.md]: 我也参考了这里：https://bitbucket.org/eigen/eigen/src/default/unsupported/Eigen/CXX11/src/Tensor/README.md
感觉也可以贴上。

File: doc/howto/dev/use_eigen_cn.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+```
+class Tensor {
+ public:
+  /*! Return a pointer to mutable memory block. */
+  template <typename T>
+  inline T* data();
+  
+  /**
+   * @brief   Return a pointer to mutable memory block.
```
Comment (typhoonzero) [inline doc/howto/dev/use_eigen_cn.md]: 因为是`_cn.md`这里的function的文档可否也替换成中文？

File: doc/howto/dev/use_eigen_cn.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+```
+paddle::framework::Tensor t;
+paddle::platform::CPUPlace place;
+// set size first
+t.Resize({2, 3});
+// allocate memory on CPU later
+t.mutable_data(place);
+```
+
+下面以AddOp为例说明Tensor的使用过程：
```
Comment (typhoonzero) [inline doc/howto/dev/use_eigen_cn.md]: 使用过程，可以加一个标题比如`### 使用Tensor`

**Comment by xinghai-sun** [comment]
Great doc!

One quick question: how to do slicing on `Eigen::Array`? (not `Eigen::Tensor` or `Eigen::TensorMap`)

e.g. `auto dims = X.dimensions()`;  how can I get `dims_new = dims[1:]`, since I need to reduce X to its first dimension with `dims_new`, e.g. `X.sum(new_dims)`.

**Comment by QiJune** [comment]
@xinghai-sun You can have a look at this https://github.com/PaddlePaddle/Paddle/pull/3774/files, could this sample meet your requirements?

File: doc/howto/dev/use_eigen_cn.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+ private:  
+  /*! holds the memory block if allocated. */
+  std::shared_ptr<Placeholder> holder_;
+  
+  /*! points to dimensions of memory block. */
+  DDim dim_;
+};
+```
+
+`Placeholder`的作用的延迟分配内存，即我们可以先定义一个Tensor，然后使用Resize接口设置Tensor的大小，最后再调用mutable_data接口分配实际的内存。
```
Comment (gongweibao) [inline doc/howto/dev/use_eigen_cn.md]: 的作用的=>的作用是

File: doc/howto/dev/use_eigen_cn.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+```
+void Compute(const framework::ExecutionContext& context) const override {
+  auto* input0 = context.Input<Tensor>("X");
+  auto* input1 = context.Input<Tensor>("Y");
+  auto* output = context.Output<Tensor>("Out");
+
+  output->mutable_data<T>(context.GetPlace());
+
+  auto X = EigenVector<T>::Flatten(*input0);
```
Comment (gongweibao) [inline doc/howto/dev/use_eigen_cn.md]: The names of variables are all lowercase. 
https://google.github.io/styleguide/cppguide.html#Variable_Names

File: doc/howto/dev/use_eigen_cn.md (hunk 1 above)
Comment (QiJune) [reply doc/howto/dev/use_eigen_cn.md]: Done

File: doc/howto/dev/use_eigen_cn.md (hunk 2 above)
Comment (QiJune) [reply doc/howto/dev/use_eigen_cn.md]: Done. 具体的例子可以参考eigen_test.cc

File: doc/howto/dev/use_eigen_cn.md (hunk 3 above)
Comment (QiJune) [reply doc/howto/dev/use_eigen_cn.md]: Done

File: doc/howto/dev/use_eigen_cn.md (hunk 4 above)
Comment (QiJune) [reply doc/howto/dev/use_eigen_cn.md]: 这里的代码注释还好吧，，也不复杂

File: doc/howto/dev/use_eigen_cn.md (hunk 6 above)
Comment (QiJune) [reply doc/howto/dev/use_eigen_cn.md]: Done

File: doc/howto/dev/use_eigen_cn.md (hunk 7 above)
Comment (QiJune) [reply doc/howto/dev/use_eigen_cn.md]: Done

File: doc/howto/dev/use_eigen_cn.md (hunk 5 above)
Comment (QiJune) [reply doc/howto/dev/use_eigen_cn.md]: Done

**Pull request merged**